ext/pcre/php_pcre.c

```diff
@@ -1314,12 +1315,16 @@ PHPAPI void php_pcre_match_impl(pcre_cache_entry *pce, zend_string *subject_str,
 							for (; i < num_subpats; i++) {
 								if (offset_capture) {
 									add_offset_pair(
-										&match_sets[i], NULL, PCRE2_UNSET, PCRE2_UNSET,
+										match_sets[i], NULL, PCRE2_UNSET, PCRE2_UNSET,
 										NULL, unmatched_as_null);
 								} else if (unmatched_as_null) {
-									add_next_index_null(&match_sets[i]);
+									zval tmp;
+									ZVAL_NULL(&tmp);
+									zend_hash_next_index_insert_new(match_sets[i], &tmp);
```

Comment:
We really should add zend_hash APIs to deal with those cases.
We have ``add_index_*()`` and ``add_assoc_*()`` when the array is wrapped in a zval but nothing when working with a HashTable directly.

Reply:
Probably, but it's worth checking first if this occurs often before adding more public APIs. I don't know off the top of my head how common this is.
In any case, I'm not gonna deal with it in this PR :p